Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-186.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 244386da5ef29a39e05d1542917fe143d0eb89d1
Containerized:False
Number of Error(s): 3
make[2]: *** [esp/src/CMakeFiles/eclwatch_fetchdeps.dir/build.make:73: esp/src/lib/package-lock.json] Error 1
make[1]: *** [CMakeFiles/Makefile2:11685: esp/src/CMakeFiles/eclwatch_fetchdeps.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-02-01_14-29-25
time logs started
Prep time   :    19 sec (00:00:19)
Build time  :   145 sec (00:02:25)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   164 sec (00:02:44)
time logs end

[Build error:](https://gist.github.com/e02ddba0ef87c6b580a60a38d7eba67b)
Build log [Build log](https://gist.github.com/HPCCSmoketest/e02ddba0ef87c6b580a60a38d7eba67b#file-result-22-02-01-14-26-26-log)

--- a/roxie/ccd/ccdmain.cpp
+++ b/roxie/ccd/ccdmain.cpp
@@ -1397,7 +1397,8 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
                         }
                         const char *soname =  roxieFarm.queryProp("@so");
                         const char *config  = roxieFarm.queryProp("@config");
-                        Owned<IHpccProtocolPlugin> protocolPlugin = ensureProtocolPlugin(*protocolCtx, soname);
+                        // NB: leaks - until we fix bug in ensureProtocolPlugin() whereby some paths return a linked object and others do not
+                        IHpccProtocolPlugin *protocolPlugin = ensureProtocolPlugin(*protocolCtx, soname);
                         roxieServer.setown(protocolPlugin->createListener(protocol ? protocol : "native", createRoxieProtocolMsgSink(ip, port, numThreads, suspended), port, listenQueue, config, certFileName.str(), keyFileName.str(), passPhraseStr.str()));
                     }
                     else

